lms/services/lti_outcomes.py

```diff
@@ -135,11 +75,7 @@ def read_result(self, outcomes_request_params):  # pylint:disable=no-self-use
             return None
 
     def record_result(  # pylint:disable=no-self-use
-        self,
-        outcomes_request_params,
-        score=None,
-        lti_launch_url=None,
-        submitted_at=None,
+        self, outcomes_request_params, score=None, **kwargs,
```

Comment:
Ok, so `outcomes_request_params` and `score` are still here but `lti_launch_url` and `submitted_at` have become `**kwargs`... They pop out again as separate `lti_launch_url` and `submitted_at` params in `_canvas_request_modification()`. So this is done to decouple `record_result()` from `_canvas_request_modification()`, to decouple it from Canvas specifics.

I wonder if `**kwargs` could be named something like `**canvas_kwargs`

Reply:
This is kind of phase 1 to pushing the caller outside of this class entirely. So the next phase is to remove the kwargs entirely and have a call-back passed in.